moto/ecr/models.py

```diff
@@ -250,17 +257,33 @@ def __init__(
 
     def _create_digest(self):
         image_manifest = json.loads(self.image_manifest)
-        layer_digests = [layer["digest"] for layer in image_manifest["layers"]]
-        self.image_digest = (
-            "sha256:"
-            + hashlib.sha256("".join(layer_digests).encode("utf-8")).hexdigest()
-        )
+        if "layers" in image_manifest:
+            layer_digests = [layer["digest"] for layer in image_manifest["layers"]]
+            self.image_digest = (
+                "sha256:"
+                + hashlib.sha256("".join(layer_digests).encode("utf-8")).hexdigest()
+            )
+        else:
+            random_sha = hashlib.sha256(
+                f"{random.randint(0,100)}".encode("utf-8")
+            ).hexdigest()
+            self.image_digest = f"sha256:{random_sha}"
 
     def get_image_digest(self):
         if not self.image_digest:
             self._create_digest()
         return self.image_digest
 
+    def get_image_size_in_bytes(self):
+        image_manifest = json.loads(self.image_manifest)
+        if "layers" in image_manifest:
+            try:
+                return image_manifest["config"]["size"]
+            except KeyError:
+                return 50 * 1024 * 1024
+        else:
+            return None
+
```

Comment:
instead of enumerating all the possible media types an image manifest may have (oci, docker, etc), I've opted here to detect if the manifest contains layer information as that implies it is an image manifest of some media type (and not a manifest list).